dali/pipeline/operators/crop/slice.cu

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) 2017-2018, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/pipeline/operators/crop/slice.h"
+
+namespace dali {
+
+template <>
+void Slice<GPUBackend>::DataDependentSetup(DeviceWorkspace *ws,
+                                           unsigned int idx) {
+  // Assumes xywh
+  const auto &input = ws->Input<GPUBackend>(idx);
+
+  // Need to copy to CPU as these values are required to perform other
+  // calculations. Far from ideal, and should be done properly once Crop op
+  // is cleaned up
+  TensorList<CPUBackend> begin;
+  begin.Copy(ws->Input<GPUBackend>(idx + 1), ws->stream());
+
+  TensorList<CPUBackend> size;
+  size.Copy(ws->Input<GPUBackend>(idx + 2), ws->stream());

```

Comment:
It may kill our perf, also we need to sync on stream if we copy something from GPU->CPU.

Reply:
I will think about this in the context of https://github.com/NVIDIA/DALI/pull/269#discussion_r232236453

It is true that right now is not very cool

Reply:
Anyway you still need to add:
```
CUDA_CALL(cudaStreamSynchronize(ws->stream()));
```
Otherwise you don't know if this copy has ended or not.

Reply:
not necessary as these inputs are now cpu always